include/nbl/asset/IImage.h

```diff
@@ -189,11 +189,11 @@ class IImage : public IDescriptor
 			{
 				return info.convert3DTexelStridesTo1DByteStrides(getTexelStrides());
 			}
-			static inline uint64_t		getLocalByteOffset(const core::vector3du32_SIMD& localXYZLayerOffset, const core::vector3du32_SIMD& byteStrides)
+			static inline uint64_t		getLocalByteOffset(const hlsl::uint32_t3& localXYZLayerOffset, const hlsl::uint32_t3& byteStrides)
 			{
-				return core::dot(localXYZLayerOffset,byteStrides)[0];
+				return hlsl::dot(localXYZLayerOffset,byteStrides);
 			}
-			inline uint64_t				getByteOffset(const core::vector3du32_SIMD& localXYZLayerOffset, const core::vector3du32_SIMD& byteStrides) const
+			inline uint64_t				getByteOffset(const hlsl::uint32_t3& localXYZLayerOffset, const hlsl::uint32_t3& byteStrides) const
```

Comment:
check if this vector is not supposed to be 4D (xyz + layer) so `uint32_t4` instead